traits/trait_list_object.py

```diff
@@ -416,6 +416,21 @@ def rename(self, name):
             self.name = name
             self.trait = trait.handler
 
+    def clear(self):
+        if self.trait.minlen > 0:
+            self.len_error(0)
+
+        if len(self) == 0:
+            return
+
+        removed = list.copy(self)
```

Comment:
`self.copy()` might look a bit more natural here. But this is likely fine, too.

Reply:
Replaced with `self.copy()` and fixed to call `list.clear` for empty lists too in 2bcd7b4.